### PR TITLE
SFS-7493: Fix not properly handling camera device with null serial

### PR DESCRIPTION
**Pull request by zzv2**
Issue discovered due to one of the RealSense cameras having a null
serial number and causing the driver to throw an exception.

### CODE REVIEW

File: src/device_utils.cpp [hunk 1]
```diff
@@ -36,10 +37,16 @@ str_map get_serial_dev_info()
         "usb",
         "usb_device");
 
-      if (parent_dev)
+      if (parent_dev != nullptr)
       {
-        std::string dev_serial = std::string(udev_device_get_sysattr_value(parent_dev, "serial"));
-        devices[dev_file_name_path] = dev_serial;
+        const auto serial = udev_device_get_sysattr_value(parent_dev, "serial");
+        if (serial) {
```
Comment (swoo-misorobotics) [inline src/device_utils.cpp:43]: do we need to do != nullptr check here?

**Review by swoo-misorobotics — APPROVED** [review]

**Review by hungrh — APPROVED** [review]

**Review by nardavin — APPROVED** [review]

File: src/device_utils.cpp (hunk 1 above)
Comment (rsinnet) [reply src/device_utils.cpp:43]: We shouldn't need to do that at all with C++11, but I don't think it hurts too much to have it.

File: src/device_utils.cpp [hunk 2]
```diff
@@ -36,10 +37,16 @@ str_map get_serial_dev_info()
...
+          std::string dev_serial = std::string(serial);
+          devices[dev_file_name_path] = dev_serial;
+        }
+        else {
+          ROS_WARN_STREAM("Found device with null serial: '" << dev_file_name_path << "'");
```
Comment (rsinnet) [inline src/device_utils.cpp:48]: Is this always going to print null or an empty string?

**Review by rsinnet — APPROVED** [review]

**Review by hjoe72 — APPROVED** [review]

File: src/device_utils.cpp (hunk 2 above)
Comment (zzv2) [reply src/device_utils.cpp:48]: no, its the `serial` variable that will be null, the `dev_file_name_path` var will have a good value like `/dev/video8`

**Pull request merged**